modules/xmpp/SdpTransformUtil.js

```diff
@@ -0,0 +1,439 @@
+ * @param {object} group the SSRC group object as defined by the 'sdp-transform'
+ * @return {Number} the primary SSRC number
+ */
+export function parsePrimarySSRC(group) {
+    return parseInt(group.ssrcs.split(" ")[0]);
+}
+
+/**
+ * Parses the secondary SSRC of given SSRC group.
+ * @param {object} group the SSRC group object as defined by the 'sdp-transform'
+ * @return {Number} the secondary SSRC number
+ */
+export function parseSecondarySSRC(group) {
+    return parseInt(group.ssrcs.split(" ")[1]);
+}
+
+function findGroup(mLine, groupSemantics, ssrcs) {
+    return mLine.ssrcGroups && mLine.ssrcGroups.find(
+            (group) => group.semantics === groupSemantics
+                            && !ssrcs || ssrcs === group.ssrcs);
+}
+
+function findGroupByPrimarySSRC(mLine, groupSemantics, primarySsrc) {
+    return mLine.ssrcGroups && mLine.ssrcGroups.find(
+            (group) => group.semantics === groupSemantics
+                            && parsePrimarySSRC(group) === primarySsrc);
+}
+
+function getSSRCCount(mLine) {
+    if (!mLine.ssrcs) {
+        return 0;
+    } else {
+        return mLine.ssrcs
+            .map(ssrcInfo => ssrcInfo.id)
+            .filter((ssrc, index, array) => array.indexOf(ssrc) === index)
+            .length;
+    }
+}
+
+/**
+ * Utility class for SDP manipulation using the 'sdp-transform' library.
+ *
+ * Typical use usage scenario:
+ *
+ * const transformer = new SdpTransformWrap(rawSdp);
+ * if (transformer.selectMedia('video)) {
+ *     transformer.addSSRCAttribute({
+ *         id: 2342343,
+ *         attribute: "cname",
+ *         value: "someCname"
+ *     });
+ *     rawSdp = transformer.toRawSdp();
+ * }
+ */
+export class SdpTransformWrap {
+
+    /**
+     * Creates new instance and parses the raw SDP into objects using
+     * 'sdp-transform' lib.
+     * @param {string} rawSDP the SDP in raw text format.
+     */
+    constructor(rawSDP) {
+        this.parsedSDP = transform.parse(rawSDP);
+        this.selectedMLine = null;
+    }
+
+    /**
+     * Returns the direction of currently selected media.
+     * @return {string} the media direction name as defined in the SDP.
+     */
+    get mediaDirection() {
+        // mLine must be selected !
+        this._assertMLineSelected();
+
+        return this.selectedMLine.direction;
+    }
+
+    /**
+     * Modifies the direction of currently selected media.
+     * @param {string} direction the new direction to be set
+     */
+    set mediaDirection (direction) {
+        // mLine must be selected !
+        this._assertMLineSelected();
+
+        this.selectedMLine.direction = direction;
+    }
+
+    /**
+     * Selects the first media SDP of given name.
+     * @param {string} mediaType the name of the media e.g. 'audio', 'video',
+     * 'data'.
+     * @return {boolean} <tt>true</tt> if the media of given type has been found
+     * and selected or <tt>false</tt> otherwise.
+     */
+    selectMedia(mediaType) {
```

Comment:
state like this also feels a bit weird, i'd rather have a util function to get an mLine and then the mLine passed to a static function to be modified